nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/InvokeScriptProcessor.java

```diff
@@ -0,0 +1,862 @@
+
+
+    public static PropertyDescriptor SCRIPT_ENGINE;
+
+    public static final PropertyDescriptor SCRIPT_FILE = new PropertyDescriptor.Builder()
+            .name("Script File")
+            .required(false)
+            .description("Path to script file to execute. Use either file or body not both")
+            .addValidator(new StandardValidators.FileExistsValidator(true))
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor SCRIPT_BODY = new PropertyDescriptor.Builder()
+            .name("Script Body")
+            .required(false)
+            .description("Body to script to execute. Use either file or body not both")
+            .addValidator(Validator.VALID)
+            .expressionLanguageSupported(true)
+            .build();
+
+
+    public static final PropertyDescriptor SCRIPT_ARGS = new PropertyDescriptor.Builder()
+            .name("Arguments")
+            .required(false)
+            .description("Arguments to pass to scripting engine")
+            .addValidator(Validator.VALID)
+            .expressionLanguageSupported(true)
+            .defaultValue("")
+            .build();
+
+    public static final PropertyDescriptor MODULES = new PropertyDescriptor.Builder()
+            .name("Module Directory")
+            .description("Path to a directory which contains modules required by the script.")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(new StandardValidators.DirectoryExistsValidator(true, false))
+            .build();
+
+    // A map from engine name to a custom configurator for that engine
+    private final Map<String, ScriptEngineConfigurator> scriptEngineConfiguratorMap = new ConcurrentHashMap<>();
+
+    private final AtomicReference<Processor> processor = new AtomicReference<>();
+    private final AtomicReference<Collection<ValidationResult>> validationResults =
+            new AtomicReference(Collections.EMPTY_LIST);
+
+    private ControllerServiceLookup controllerServiceLookup;
+    private String initializationContextId;
+
+    private final AtomicBoolean isInitialized = new AtomicBoolean(false);
+    private final Lock lock = new ReentrantLock();
+    private SynchronousFileWatcher scriptWatcher;
+
+    private Map<String, ScriptEngineFactory> scriptEngineFactoryMap;
+    private ScriptEngineManager scriptEngineManager;
+    private ScriptEngine scriptEngine;
+    private String scriptEngineName;
+    private String scriptPath;
+    private String modulePath;
+    private String scriptBody;
+
+    private ScheduledExecutorService reloadService = null;
+
+    private List<PropertyDescriptor> descriptors;
+
+    /**
+     * Initializes this processor
+     *
+     * @param context in which to perform initialization
+     */
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        initializationContextId = context.getIdentifier();
```

Comment:
Ah, I wasn't sure if the initialization context's identifier and lookup would be the same as what the superclass would return